Add kali distro

KyleEvers opened the pull request:
Add Kali distro to upgrade by using shell commands instead of ansible apt module due to a bug causing the apt module to hang and never complete.

KyleEvers committed:
add update commands for kali

KyleEvers committed:
move update to before the reinstall of initramfs

KyleEvers committed:
add auto yes to reinstall command

KyleEvers committed:
fix comments for tasks

KyleEvers committed:
testing moving environment variable to the end of the ansible command

KyleEvers committed:
remove quotes around environment variable value

KyleEvers committed:
switch from shell commands to ansible apt

KyleEvers committed:
alphabetize distro orders, change line length to pass yaml lint, remove duplicate task

KyleEvers committed:
add kali to support distros for molecule

KyleEvers committed:
testing shell command again because apt hangs

KyleEvers committed:
remove environment variable because it is set by default in ansible, add cache to aviod multiple apt-get updates

KyleEvers committed:
change to using upgrade instead of name for ansible apt

KyleEvers committed:
ansible can't find dist for kali upgrade so trying yes

KyleEvers committed:
switch back to using shell command and not ansible apt module. See comment about issue for more info

KyleEvers committed:
ignore ansible lint to avoid use of shell over apt module

KyleEvers committed:
fix spacing on comment

KyleEvers committed:
set warning to false because a warning is thrown for using shell apt instead of ansible apt module

KyleEvers committed:
add warn false as an arg

KyleEvers committed:
add no test for molecule-idempotence-notest

KyleEvers committed:
add tags for molecule-idempotence-notest to skip idempotence which fails for apt-get update via shell

felddy commented on tasks/main.yml:
```diff
@@ -14,7 +14,34 @@
+    warn: false
+  environment:
+    DEBIAN_FRONTEND: noninteractive
+  tags: molecule-idempotence-notest
```
Why no idempotence test?  Add a comment.

KyleEvers replied:
Addressed in commit 9f51229

jsf9k replied:
`shell` is not necessary here either.  `command` will work and should allow you to get rid of at least one `noqa`.

jsf9k commented on tasks/main.yml:
```diff
@@ -14,7 +14,34 @@
+- name: Reinstall cryptsetup-initramfs so upgrade doesn't fail
+  apt:
+    name: 'cryptsetup-initramfs'
+    state: latest  # noqa 403
```
You should add a comment above this line stating what this mysterious code does.  Same thing below where you use the magic incantation `noqa 303 305`.

KyleEvers replied:
Addressed in commit 9f51229

jsf9k replied:
Why is "latest" necessary here if you're only installing a single package?  Seems it should work with `present` instead of `latest`.  (BTW, `present` is the default.)

felddy reviewed (approved):
Approved... but see comment about comments.

dav3r commented on tasks/main.yml:
```diff
@@ -14,7 +14,34 @@
+  when: ansible_distribution_release == "kali-rolling"
+
+- name: apt-get update
+  shell: apt-get update  # noqa 303 305
```
Please add a brief comment explaining why you are skipping checks 303 and 305 here.  Same goes for line 38 below.

KyleEvers replied:
Addressed in commit 9f51229

jsf9k replied:
You don't need `shell` here since you aren't using any shell features (like file redirection).  `command` will work, and should let you get rid of at least one of the `noqa`s.

jsf9k commented on tasks/main.yml:
```diff
@@ -14,7 +14,34 @@
+  args:
+    warn: false
+  when: ansible_distribution_release == "kali-rolling"
+  tags: molecule-idempotence-notest
```
Is this some molecule-fu that tells the idempotence test to ignore for this step?  If so then there should be a comment explaining that.

KyleEvers replied:
Yes, it is molecule-fu to ignore the idempotence test. Addressed in commit 9f51229

jsf9k replied:
Typo in "guarantee".

jsf9k commented on tasks/main.yml:
```diff
@@ -14,7 +14,34 @@
+    warn: false
+  environment:
+    DEBIAN_FRONTEND: noninteractive
+  tags: molecule-idempotence-notest
```
See previous comment regarding `molecule-idempotence-notest`.

KyleEvers replied:
Addressed in commit 9f51229

dav3r commented on tasks/main.yml:
```diff
@@ -14,7 +14,34 @@
+- name: Reinstall cryptsetup-initramfs so upgrade doesn't fail
+  apt:
+    name: 'cryptsetup-initramfs'
+    state: latest  # noqa 403
```
Please add a bit more explanation to your comment above about why we want to use `latest` here and above.

KyleEvers replied:
Addressed in commit 9f51229

dav3r reviewed (approved):
Looks good, but please flesh out your comments a bit more (see my comments).
👍

jsf9k commented on tasks/main.yml:
```diff
@@ -14,7 +14,34 @@
+  shell: apt-get update  # noqa 303 305
+  args:
+    warn: false
+  when: ansible_distribution_release == "kali-rolling"
```
Is `ansible_distribution_release` really the only way to tell that this is Kali?  What is `ansible_distribution` for Kali?

KyleEvers replied:
ansible_distribution are predefined ansible values and Kali is not one of them.

jsf9k reviewed (changes requested):
Requested some minor changes.

KyleEvers committed:
add comments based on feedback from PR

KyleEvers committed:
fix typo in comment

jsf9k committed:
Remove "latest" since "present" is good enough here (and idempotent)

jsf9k committed:
Prefer command to shell, since we aren't using any shell features

jsf9k committed:
Correct typos in comments

jsf9k committed:
Merge remote-tracking branch 'origin/add_kali_distro' into add_kali_distro

jsf9k committed:
Fix typo that @KyleEvers committed but I foolishly obliterated when I merged in his changes

jsf9k committed:
Wrap all the Kali linux code in a block

This allows us to apply the when clause once instead of multiple
times.

Also reorder the update and cryptsetup-initramfs installation steps,
so that we only need to update the package cache a single time.
